### PR TITLE
Optimize `remove_unreachable_nodes` in `gensim.summarization`

### PR DESCRIPTION
**Pull request by horpto**

### CODE REVIEW

File: gensim/summarization/commons.py [hunk 1]
```diff
@@ -63,5 +63,5 @@ def remove_unreachable_nodes(graph):
     """
 
     for node in graph.nodes():
-        if sum(graph.edge_weight((node, other)) for other in graph.neighbors(node)) == 0:
+        if all(graph.edge_weight((node, other)) == 0 for other in graph.neighbors(node)):
```
Comment (horpto) [inline gensim/summarization/commons.py]: Can we also delete null-weighted edges for reachable nodes?
@piskvorky  @menshikh-iv

Comment (piskvorky) [reply gensim/summarization/commons.py]: A question for @fedelopez77, who is the author of this module :)

Comment (fedelopez77) [reply gensim/summarization/commons.py]: @horpto I think that it would be reasonable, but test it first.
For the corner case that all edges are zero, according to (https://github.com/RaRe-Technologies/gensim/blob/1e11073196701adb6fabb8ccd5f7689db01dc38c/gensim/summarization/summarizer.py#L362-L363) this is executed before: 
https://github.com/RaRe-Technologies/gensim/blob/1e11073196701adb6fabb8ccd5f7689db01dc38c/gensim/summarization/summarizer.py#L105-L106

**Comment by menshikh-iv** [comment]
For merging this PR, @horpto please add tests for case suggested by @fedelopez77 (and merge fresh `develop` too).

File: gensim/summarization/commons.py (hunk 1 above)
Comment (horpto) [reply gensim/summarization/commons.py]: Ok,  actually we don't need to delete null-weighted edges, as:

- we can not add such edge at all
- in pagerank we are passing by all edges anyway

As another note, I can notice that gensim summarizer has a big serious memory problems ([this](https://stackoverflow.com/questions/50567108/gensim-summarizer-throws-memoryerror-any-solution) and [this](https://stackoverflow.com/questions/50516951/python-extracting-text-and-summarize-multiple-file-types-in-directory-with-tik)).

**Comment by horpto** [comment]
@menshikh-iv review, please

**Comment by menshikh-iv** [comment]
Thanks @horpto :+1:
